NDTensors/ext/NDTensorscuTENSORExt/contract.jl

```diff
@@ -0,0 +1,93 @@
+using NDTensors:
+  NDTensors,
+  BlockSparseTensor,
+  Tensor,
+  array,
+  contract!,
+  contraction_output,
+  contraction_output_type,
+  contract_inds,
+  dense,
+  inds
+using NDTensors.Expose: Exposed, expose, unexpose
+using cuTENSOR: CuArray, CuTensor
+
+function NDTensors.contract(
+  Etensor1::Exposed{<:CuArray},
+  labelstensor1,
+  Etensor2::Exposed{<:CuArray},
+  labelstensor2,
+  labelsoutput_tensor,
+)
+  tensor1 = unexpose(Etensor1)
+  tensor2 = unexpose(Etensor2)
+
+  return cutensor_contract(
+    tensor1, labelstensor1, tensor2, labelstensor2, labelsoutput_tensor
+  )
+end
+
+## working to fix blocksparse implementation. 
+function NDTensors.contract(
```

Comment:
This function is necessary so no error is thrown when trying to contract blocksparse tensors with cutensor enabled.  It does not however use cuTENSOR based contraction because there is no easy way to convert `BlockSparseTensors` to `CuTensors` and simply uses the previous `CUDA.mul!` function.  I am currently working on a solution for this.

Reply:
I think we should only overload `contract` for `DenseTensor` in the `cuTENSOR` extension. `BlockSparseTensor`s will get contracted as usually, but when it goes to perform the contractions over blocks, if the blocks have `Dense` storage they will get sent to `cuTENSOR`, but that logic will work automatically and doesn't have to be defined explicitly in the package extension.

Reply:
I am closing this comment because I am limiting the code to overload `DenseTensor` and moving it down to the `contract!` level so the code will work with blockwise contractions.